### ISSUE TITLE
Allow different regions in AWS resource creation functions

### ISSUE THREAD

**Issue by fdalmaup** [wazuh/qa-integration-framework#150]
# Description

Part of #133 was to make the necessary modifications to be able to create resources across multiple regions due to the `test_regions` tests. Currently, the framework generates a client to connect to AWS using the `us-east-1` region, therefore if we want to generate and fetch events from other regions the region would need to be configurable, according to the test case being executed.

**Comment by fdalmaup** [comment]
# Issue Update

- Modified the FW code to receive the client according to the service as an argument of the different functions found in `src/wazuh_testing/modules/aws/utils.py`. This required migrating the `boto3` client creation and management from the FW to the `conftest.py` file of the tests as fixtures. The execution of the `test_discard_regex.py` tests was successful:
```console
root@vagrant:/# pytest -x /wazuh/tests/integration/test_aws/test_discard_regex.py
=========================================================================== test session starts ===========================================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.4.0
rootdir: /wazuh/tests/integration, configfile: pytest.ini
plugins: metadata-3.1.1, html-3.1.1
collected 17 items                                                                                                                                                        

wazuh/tests/integration/test_aws/test_discard_regex.py .................                                                                                            [100%]

===================================================================== 17 passed in 374.45s (0:06:14) ======================================================================
```

- The changes in the bucket functions that use the hardcoded `us-east-1` default region are in progress.

**Comment by fdalmaup** [comment]
# Issue Update

- Modifications in the `src/wazuh_testing/modules/aws/utils.py` file and  `src/wazuh_testing/modules/aws/data_generator.py` file methods to take the region as a parameter and generate the file path and filename according to the required service. 
- The tests keep passing, using multiple regions were tested for the buckets and CloudWatch Logs services in the functioning `test_discard_regex` cases:
  - For the buckets' cases, the client is only instantiated once and the different `regions` found in the metadata are used for the file paths and filenames.
  - For the CloudWatch Logs case, a client is instantiated per region, and logs are generated in the corresponding region.

- Due to the recently merged #151, a rebase of the dev branch is the only task remaining to open the PR.
